### PR TITLE
Change GroupListing to support mixed content

### PR DESCRIPTION
**Pull request by anastasop**
This is to support mixed content for groups.

I am not very fond of the model GroupEntry where `group` or `asset` maybe empty. However we currently use swagger 2 which does not have good support for polymorphic types (fixed in swagger 3) I am open to suggestions

### CODE REVIEW

**Review by alekshs — APPROVED** [review]

**Comment by anastasop** [comment]
I added the new endpoint for the group browser. @alekshs @antalakas Please have a look.

I added to this PR because i wanted it to be reviewed together with the new GroupListing definition

**Comment by anastasop** [comment]
Added some more changed based on discussion with @alekshs

**Review by alekshs — APPROVED** [review]

**Comment by anastasop** [comment]
@alekshs @antalakas I changed the definition of Group path after a discussion with Alex. I think that the PR is now complete and reflects what UI expects from the rest server

**Review by alekshs — APPROVED** [review]
lgtm

**Comment by anastasop** [comment]
@alekshs @antalakas Added to group attributes `last_accessed` `share_count` `allowed_actions` and the endpoint for batch operation. Please have a look. With these additions i think the group API is complete. It remains to add the share endpoint which will be added after the rest server PR is merged

File: openapi-v1.yaml [hunk 1]
```diff
@@ -2737,11 +2751,35 @@ definitions:
...
+        type: integer
+        x-omitempty: false
+      ancestors:
+        description: The ancestors of the group. First is the top level group
+        type: array
+        items:
+          $ref: "#/definitions/GroupAncestor"
+        x-omitempty: false
+      share_count:
+        description: number of unique namespaces this array is shared with
```
Comment (antalakas) [inline openapi-v1.yaml:2765]: this group is shared with

File: openapi-v1.yaml [hunk 2]
```diff
@@ -6379,7 +6579,7 @@ paths:
         200:
           description: the group contents
           schema:
-            $ref: '#/definitions/GroupListing'
+            $ref: '#/definitions/GroupContents'
```
Comment (antalakas) [inline openapi-v1.yaml:6582]: Should this be `GroupBrowserData`?

File: openapi-v1.yaml [hunk 3]
```diff
@@ -2772,27 +2810,65 @@ definitions:
...
+      asset:
+        description: If not empty, this entry is an asset
+        $ref: "#/definitions/ArrayInfo"
+        x-omitempty: true
+      group:
+        description: If not empty, this entry is a group
+        $ref: "#/definitions/Group"
+        x-omitempty: true
+
+  GroupContents:
```
Comment (antalakas) [inline openapi-v1.yaml:2826]: This definition is only used in `public` endpoint. But I also think that public should return `GroupContents` like shared and owned (see next comment)

**Review by antalakas — CHANGES_REQUESTED** [review]

File: openapi-v1.yaml (hunk 1 above)
Comment (anastasop) [reply openapi-v1.yaml]: :+1:

**Comment by anastasop** [comment]
@alekshs @antalakas A new commit to support multiple names for groups. The name moves from the url to the body

**Comment by anastasop** [comment]
Closing. Superseded by https://github.com/TileDB-Inc/TileDB-Cloud-API-Spec/pull/305

**Pull request closed without merging**